### PR TITLE
refactor: Update ruff and enable pyupgrade rules

### PR DESCRIPTION
**Pull request by hoxbro**
Bump ruff version and enable [pyupgrade](https://docs.astral.sh/ruff/rules/#pyupgrade-up) rules

### CODE REVIEW

File: datashader/datashape/coretypes.py [hunk 1]
```diff
@@ -26,7 +25,7 @@ class Type(type):
     _registry = {}
 
     def __new__(meta, name, bases, dct):
-        cls = super(Type, meta).__new__(meta, name, bases, dct)
+        cls = super(Type, meta).__new__(meta, name, bases, dct)  # noqa: UP008
```
Comment (hoxbro) [inline datashader/datashape/coretypes.py:28]: I'm not entirely sure this is a safe fix, so I just ignored it.

File: datashader/tests/test_colors.py [hunk 2]
```diff
@@ -15,10 +15,10 @@ def test_hex_to_rgb():
 
 
 def test_rgb():
-    assert rgb(u'#FAFBFC') == (250, 251, 252)
     assert rgb('#FAFBFC') == (250, 251, 252)
+    assert rgb('#FAFBFC') == (250, 251, 252)
+    assert rgb('blue') == (0, 0, 255)
```
Comment (hoxbro) [inline datashader/tests/test_colors.py:20]:
```suggestion
```
Now duplicate

File: datashader/datashape/internal_utils.py [hunk 3]
```diff
@@ -76,7 +75,7 @@ def _toposort(edges):
     """
     incoming_edges = reverse_dict(edges)
     incoming_edges = dict((k, set(val)) for k, val in incoming_edges.items())
-    S = set((v for v in edges if v not in incoming_edges))
+    S = set(v for v in edges if v not in incoming_edges)
```
Comment (hoxbro) [inline datashader/datashape/internal_utils.py:78]:
```suggestion
    S = {v for v in edges if v not in incoming_edges}
```

**Review by jbednar — APPROVED** [review]
Nice cleanups, thanks!

**Pull request merged**